pkg/kubelet/cm/container_manager_linux.go

```diff
@@ -365,14 +365,28 @@ func (cm *containerManagerImpl) NewPodContainerManager() PodContainerManager {
 			enforceCPULimits:  cm.EnforceCPULimits,
 			// cpuCFSQuotaPeriod is in microseconds. NodeConfig.CPUCFSQuotaPeriod is time.Duration (measured in nano seconds).
 			// Convert (cm.CPUCFSQuotaPeriod) [nanoseconds] / time.Microsecond (1000) to get cpuCFSQuotaPeriod in microseconds.
-			cpuCFSQuotaPeriod: uint64(cm.CPUCFSQuotaPeriod / time.Microsecond),
+			cpuCFSQuotaPeriod:   uint64(cm.CPUCFSQuotaPeriod / time.Microsecond),
+			podContainerManager: cm,
 		}
 	}
 	return &podContainerManagerNoop{
 		cgroupRoot: cm.cgroupRoot,
 	}
 }
 
+func (cm *containerManagerImpl) HasExclusiveCPUs(pod *v1.Pod) bool {
+	for _, container := range pod.Spec.Containers {
```

Comment:
I think this _fine_ with me, but I don't fully remember _exactly_ how init contianers work with exclusive cpus, but I think here its pretty important to be consistent.

I think I'm happy with a pod where;
```
- Pod is A
  - initContainer B does NOT have exclusive CPU (cpu request+limit: 20m)
  - container C has exclusive CPU (cpu request+limit: 1)
```

to have;
```
- Pod level A has NO cpu limit
- initContainer B does have a cpu limit
- container C has NO cpu limit
```

However, I think in this state the current implementation here will also have no cpu limit on the init container even though it has a cpu limit. I think that is because the container level resource check for the init container is using this function that only loops over the actual container, and not init ones.

Sorry if this is already discussed.

Reply:
Right, very good point about init containers. Need to have another pass about it and fully flesh it out.

Reply:
surely we will need to treat sidecar containers as app containers and extend `HasExclusiveCPUs` to deal with that (cc @SergeyKanzhelev )

Reply:
commented in #70585, this is done matching existing logic of cpumanager `podGuaranteedCPUs` / `guaranteedCPUs`